R/backend-mssql.R

```diff
@@ -93,7 +93,10 @@ simulate_mssql <- function(version = "15.0") {
 #' @export
 `sql_query_insert.Microsoft SQL Server` <- function(con, x_name, y, by, ...,
                                                     conflict = c("error", "ignore"),
-                                                    returning_cols = NULL) {
+                                                    returning_cols = NULL,
+                                                    method = NULL) {
+  method <- method %||% "where_not_exists"
+  arg_match(method, "where_not_exists", error_arg = "method")
```

Comment:
I wonder if it's worth just leaving this unchecked so that the user can use whatever the database supports. Otherwise I worry that we'd need to keep this in sync with database version, which is going to be quite a bit of work for little gain.

Reply:
I thought about leaving this unchecked but I don't really see how this is going to work.

Let's say someone wants to support `method = "fancy"` for Postgres. Then `sql_query_insert.PqConnection()` needs to know about `method = "fancy"` to generate the correct SQL for this, i.e. we need something like this:

```r
sql_query_insert.PqConnection <- function(...) {
  if (method == "fancy") {
    sql_query_insert_fancy(...)
  }

  ...
}
```

Reply:
Ooooh, I just thought this affected the value of some component of the generated SQL.